drivers/SmartThings/zwave-switch/src/leviton-smart-switch/init.lua

```diff
@@ -0,0 +1,114 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local Configuration = (require "st.zwave.CommandClass.Configuration")({ version = 4 })
+local preferences = require "preferences"
+local cc = require "st.zwave.CommandClass"
+
+local LEVITON_MANUFACTURER_ID = 0x001D
+local LEVITON_PRODUCT_TYPE_ZWXXX = 0x0002
+local LEVITON_PRODUCT_ID_ZW6HD = 0x0041
+local LEVITON_PRODUCT_ID_ZW15S = 0x0042
+
+local function can_handle_leviton_zwxxx(opts, driver, device, ...)
+  return device:id_match(
+    LEVITON_MANUFACTURER_ID,
+    {LEVITON_PRODUCT_ID_ZW6HD, LEVITON_PRODUCT_ID_ZW15S},
+    LEVITON_PRODUCT_TYPE_ZWXXX
+  );
+end
+
+local function handle_configuration_update(self, device, cmd)
+
+  local parameters = preferences.get_device_parameters(device)
+  local target_parameter_number = cmd.args.parameter_number
+  local reported_value = cmd.args.configuration_value
+  local target_parameter_name = nil
+
+  if not parameters then
+    return
+  end
+
+  for id, parameter in pairs(parameters) do
+    if parameter.parameter_number == target_parameter_number then
+      target_parameter_name = id
+      break
+    end
+  end
+
+  if target_parameter_name == nil then
+    return
+  end
+
+  device:set_field(target_parameter_name, reported_value)
+  device:set_field(target_parameter_name, reported_value, {persist = true})
+
+  -- FIXME: SmartThings Mobile App UI does not sync with the updated value
+  -- https://community.smartthings.com/t/st-edge-driver-for-leviton-z-wave-plus-dimmers-dz6hd-dz1kd/257938/4
+
+end
+
+local function device_init(self, device)
+
+  local parameters = preferences.get_device_parameters(device)
+
+  if parameters then
+    for id, pref in pairs(parameters) do
+      device:send(Configuration:Get({
+        parameter_number = pref.parameter_number
+      }))
+    end
+  end
+end
```

Comment:
We have had users report a race condition where the hub might not be ready for z-wave transmission during the `init` lifecycle event. We haven't been able to reproduce this locally, but I'd be careful about it. Since the values you read are never used, it does not seem particularly crucial in any event.